code/core_save.c

```diff
@@ -57,7 +61,8 @@ void save_galaxies(int filenr, int tree)
       // only open the file if it is not already open.
       if( !save_fd[n] )
 	    {
-        snprintf(buf, MAX_STRING_LEN - 1, "%s/%s_z%1.3f_%d", OutputDir, FileNameGalaxies, ZZ[ListOutputSnaps[n]], filenr);
+        snprintf(buf, MAX_BUF_SIZE - 1, "%s/%s_z%1.3f_%d", OutputDir, FileNameGalaxies, ZZ[ListOutputSnaps[n]], filenr);
```

Comment:
Since `buf` is allocated as `MAX_STRING_LEN`, the second parameter should be `MAX_STRING_LEN - 1`

Reply:
hi Manodeep. You're right that `buf` is declared to be smaller. But I would rather fix that by changing line 22
````
char buf[MAX_STRING_LEN];
````
to
````
char buf[MAX_BUF_SIZE];

````
because there are two potentially long strings - `OutputDir` and `FileNameGalaxies` - plus a bit of extra stuff that are written to `buf`.

Reply:
Yup - good call